parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/state/ProcessingShard.java

```diff
@@ -94,10 +103,16 @@ public long getCountWorkInFlight() {
     }
 
     public WorkContainer<K, V> remove(long offset) {
+        // from onPartitionsRemoved callback, need to deduce the available worker count for the revoked partition
+        WorkContainer<K, V> toRemovedWorker = entries.get(offset);
+        if (toRemovedWorker != null && toRemovedWorker.isAvailableToTakeAsWork()) {
+            dcrAvailableWorkContainerCntByDelta(1);
+        }
         return entries.remove(offset);
     }
 
 
+
     // remove staled WorkContainer otherwise when the partition is reassigned, the staled messages will:
     // 1. block the new work containers to be picked and processed
     // 2. will cause the consumer to paused consuming new messages indefinitely
```

Comment:
refactor `removeStaleWorkContainersFromShard` to decrement `availableWorkContainerCnt` similarly to the `public WorkContainer<K, V> remove(long offset)` method above

Reply:
fixed

Reply:
Similarly in the `ShardManager` class `removeStaleContainers()` - should remove stale containers from the `retryQueue` as well.